src/components/renderer/form-masked-input.vue

```diff
@@ -76,38 +78,44 @@ export default {
     getUserDateFormat,
     getUserDateTimeFormat,
     convertToData(newValue) {
-      switch (this.dataFormat) {
-        case 'string':
-          newValue = newValue.toString();
-          break;
-        case 'boolean':
-          newValue = Boolean(newValue);
-          break;
-        case 'currency':
-        case 'percentage':
-        case 'float':
-          newValue = parseFloat(newValue);
-          break;
-        case 'int':
-          newValue = parseInt(newValue);
-          break;
-        case 'date':
-          if (this.componentName === 'FormDatePicker') {
-            newValue = moment.utc(newValue, [getUserDateFormat(), moment.ISO_8601], true).toISOString().split(RegExp('T[0-9]'))[0];
-          }
-          break;
-        case 'datetime':
-          if (this.componentName === 'FormDatePicker') {
-            newValue = moment(newValue, [getUserDateTimeFormat(), moment.ISO_8601], true).toISOString();
-          }
-          break;
-        case 'array':
-          break;
-        default:
-          newValue = newValue.toString();
-          break;
+      if (this.customFormatter) {
+        newValue = newValue.replace(/[^\w]/g, '');
+        return newValue;
+      } else {
```

Comment:
As there is a return command inside the if-block, to have an else-block  is not necessary. Furthermore, we'll reduce the code nesting

Reply:
Good Catch! I'll fix it!